### DISCUSSION TITLE
How to retrieve memberslingbox details via Postman

### DISCUSSION THREAD

**Discussion by rbf121** [GerryDazoo/Slinger#389] [General]
As of 11/30/2022, the Sling APIs to retrieve your memberslingbox details are still online.  You can't get them from the original URL we were using but the underlying APIs are still active.

You need to call 2 APIs in order to get the memberslingbox details.  

You can use Postman to create the 2 required requests.

**Request 1:**

POST https://secure.sling.com/slingaccounts/rest/v3/identitypoint?contentType=application/json

Under Body select x-www-form-urlencoded
Enter following key / values

| key | value |
| - | - |
| context | SLINGIT_PORTAL |
| policy_code | MAS |
| email_address | {user-email} |
| password | {user-password} |

Hit send.  If it worked, you should get a 200 OK with JSON response.  Save the identity and ticket from response, you will need those for the next call.

**Request 2:**

New Request
GET https://secure.sling.com/slingaccounts/rest/v5/member/{member-id}/device/slingboxes?contentType=application/json

Replace {member-id} with the "identity" value from first request

Under Authorization tab, select Basic Auth.
Enter the "ticket" value from first request as the username.  Leave the password blank.
Hit Send.

It should list out your memberslingbox JSON.

Thanks @socelillyhammer for pointing out that the slingplayer desktop auth was still working.  Hopefully this is helpful for those that need to recover their account box info and/or can't install slingplayer desktop for whatever reason.

**Comment by bradut** [comment]
Thank you for this.
Unfortunately I've got a 423 Locked response at the POST request. 
I hope others will be more lucky.

**Reply by socelillyhammer** [reply]
I tried it.  Worked for me.  If you need your SB information send me you password and login info and I'll do the legwork to pull it for you.

**Reply by carranzafp** [reply]
same for me, response 423 Locked.   But 2 days ago I was able to secure my passwords with the Desktop application

**Reply by rbf121** [reply]
How quickly did it come back with a 423 Locked Response?  Have you tried it again since getting that response?  I noticed a few requests were taking a while but I would just cancel and try again which went through.

**Reply by m43kwon** [reply]
I also get 423 locked, using either Postman or curl. Response is immediate.

**Comment by DoogieHowitzerWMD** [comment]
Wow, that's a great solution. We can pull that info even while the slingbox stays blocked from outside communications.

This really needs to be added to the readme to replace the info there that no longer works post-shutdown.

**Edit:** I've made a pull request with this method added to the readme.

**Reply by rbf121** [reply]
Maybe hold off on doc updates until other people actually confirm that this is working for them.  Also, I'm not sure how well Postman instructions would work for the masses.  You can share postman collections which might be easier for people to use.

**Comment by carranzafp** [comment]
2 days ago I was able to get the console.log from the Desktop player so I have my passwords secured by now.  Today I tried this postman method and on the first step (POST) I am getting response "423Locked (WebDAV) (RFC 4918)".  I am using my email and password.  

If for some reason I use wrong password it only sits there until the request expire

**Comment by rbf121** [comment]
I'm thinking a couple curl commands might be easier for most people:
open cmd window.
Run:
`curl -X POST https://secure.sling.com/slingaccounts/rest/v3/identitypoint?contentType=application/json -H "Content-Type: application/x-www-form-urlencoded" -d "context=SLINGIT_PORTAL&policy_code=MAS&email_address=&password="`

Enter the email and password after each equal sign above.

**Sample output:**
{"identity":1234567,"ticket":"1535d659-7354-4096-80d1-f689252630be-prod"}

For second request, you can open a web browser and enter
https://secure.sling.com/slingaccounts/rest/v5/member/{member-id}/device/slingboxes?contentType=application/json

Replace {member-id} with "identity" output from first curl.  When opening from a browser, it will prompt for user/pw.  Enter the "ticket" as username and blank password

Or use curl for second request:
Example on Windows:
curl -u 1535d659-7354-4096-80d1-f689252630be-prod: https://secure.sling.com/slingaccounts/rest/v5/member/1234567/device/slingboxes?contentType=application/json

On Linux (put in single quotes, including a space after : ):
curl -u '1535d659-7354-4096-80d1-f689252630be-prod: ' https://secure.sling.com/slingaccounts/rest/v5/member/1234567/device/slingboxes?contentType=application/json

Replace 1535d659-7354-4096-80d1-f689252630be-prod with "ticket" from first request.  Replace 1234567 with "identity" from first request

**Reply by tryin100** [reply]
I tried your method above (Run:) line worked fine.  For second request, when using the "identity" number as the "member id", i got "The server understood the request, but is refusing to fulfill it"

**Comment by tojannaon** [comment]
I have a bricked Solo and had not retrieved my password before November 9. The Postman method worked for me, with thanks @rbf121. Now just hoping that a way can be figured out to unbrick the Solo.

**Reply by GerryDazoo** [reply]
Don't hold your breath. We're busy trying to keep the unbricked boxes alive. Right now we have no idea if the Solo and Pro boxes are unbrickable.

**Comment by tryin100** [comment]
using the postman approach what does one enter for email and password?  the email one had used normally with slingbox and the decrypted password?  I tried using my email that I normally used with slingbox and the password I had normally used, then with the decrypted password that i now use with (working) slinger.    The postman approach didn't work for me

**Reply by rbf121** [reply]
Your sling user email address and the password that you set up in your account with sling.  Basically, the same user/pw you used to connect to slingbox.com for example.  

You do not need your decrypted password at all for this.  The whole point of this is in order to get the admin passwords for each slingbox device.  If you already have those passwords then you don't need to do this.

Note: there is an updated readme as part of the code now which I think does a better job of explaining the postman part so you may want to refer to that now.

**Comment by davepins0n** [comment]
Finally got everything from Sling using curl, it took quite a while to get the prompt to login with the user... now on to see if can use my boxes again. Thanks!!!!

**Comment by mosher1313** [comment]
I assume this way in no longer working? Has anybody had any recent success with this? I just tried Postman and CMD prompt and had no luck.

**Reply by GerryDazoo** [reply]
I just gave it a try and it worked OK for me.  I used the "curl" commands documented on Dec 2 above. Just saying it doesn't work isn't very helpful. Please post all the output from your session.

**Reply by mosher1313** [reply]
Thanks for checking. It just hangs and times out with "curl: (56) Send failure: Connection was reset" for me when using curl, which made me think it was no longer fully functioning. Perhaps it's due to me not remembering my password correctly. Knowing it does indeed work still I will try troubleshooting a bit more.

Appreciate the help.

**Reply by mosher1313** [reply]
Interestingly, adding --verbose to my same "curl" command actually gave me the output and I was able to access my information. Thank you!

**Comment by jcdmorris** [comment]
Confirming the Postman method was successful for me on 15-Feb-2023.

**Comment by waelelfeqy** [comment]
postman method is not working for me Jan 7th 2024

**Reply by bradut** [reply]
I think it only worked only for a few more months after Sling service shutdown in Nov.2022 and only for some users (didn't work for me).
To use the Slinger server, I didn't have to use the original password.

**Comment by waelelfeqy** [comment]
I have the M1 and I owned it since 2015 when I bought it at a discount from Tigerdirect before they went down. I registered the darn thing and then I rarely really used it because when I had bought it originally, I thought it came with a TV tuner and obviously M1 doesn't come with a tuner and Tigerdirect was gone. fast forward a few weeks ago, I rushed into resetting it and you know the rest. I used The slinger server and it accepts the admin as password, but if I rebooted the slingbox for any reason, the password changes to whatever. When I try to stream via the slinger server, it seems as if it connects, but then the slingbox refuses the connection and doesn't stream anything. So, I don't know if my M1 is bricked forever or there is a way out. I hate to trash a good device that still could work just fine till that day. I'm considering buying a Pro HD from Ebay, since it has a tuner, as I always wanted, but then I don't know if buying a used one with the password included be better or buying a new one in the box, any thoughts ?
________________________________

**Reply by bradut** [reply]
I have a PRO-HD and I know for sure  (from this discussion list and confirmed personally) that it always can be factory reset {user: admin, password: admin}, so if you buy a used one, you don't have to worry about its previous login info.

About M1, I am not sure, please search here, as some types of Slingboxes cannot recover after being "poisoned" by Sling Media (Dish Network).<br />
>No that is not the case. We have a procedure now to totally unbrick the ProHD with the 1.5 software. Unfortunately it won't configure the 350/500/M1
https://github.com/GerryDazoo/Slinger/discussions/350 
<br />
 
To resuscitate a PRO-HD SlingBox. I would recommend this guide:
https://www.mediafire.com/file/ppw6ep09t99xhu0/Slingbox_ProHD_PC_Unbricking-Setup-Network-Guide.pdf/file
I guess they provide a link to slingplayer 1.5.xxx, if not, you can search for it on this website and will find links to download it.

Also, you may have a look in this thread (and any similar ones):
https://github.com/GerryDazoo/Slinger/discussions/357

**Reply by flipper2222222** [reply]
this is the same for  SLINGBOX PRO SB200-100 - the  red box, worked for me personally

**Comment by waelelfeqy** [comment]
Thanks for your reply, I appreciate it. Actually, my M1 is Un-resettable, but that's ok because I was able to find the old password from my old hard drive under users/me/appdata/roaming/sling media. After copying files from the old drive to my current windows 10 system, sling player 5.0083 now works and I can see the stream just fine. In the case of M1, if you can't find the old password, then it is bricked, at least in my experience. Even slinger server will not be able to work without the password. Now, I can watch locally on my network and with a sling player installed on each pc, or use slinger server and port forward and watch from internet; not bad !!! question is, can we access the box from outside the network without the slinger server ? I know already the ip address and default port for M1 (5301) but when I try to open that address with VLC, it just fails or times out. I used wireshark to trace the packages from and to pc and slingbox, but nothing useful to me to access the slingbox without the sling player installed locally or the slinger server streaming over the internet. I wasted a whole day trying to figure out the URL for the slingbox that would work with vlc, like: ***@***.*** ip:5301; ***@***.*** ip:5301; or add slingbox after that address or slingbox/live....etc and other iterations, but to no avail. Why do I want this ? to be able to access the slingbox from outside network without having to run a server on my main pc. I'm hoping somebody might have an idea, but for time being, at least, I salvaged the M1, thanks to all of you guys.

________________________________

**Reply by bradut** [reply]
I am glad to learn you can access your M1 !
I do hope that you have **blocked the outgoing port(s) in your network**, so that your slingbox cannot phone home to receive the poison pill.

With respect to your concerns related to accessing your slingbox form outside without Slinger player:
SlingBox devices broadcast a **proprietary** encoded stream, which can be consumed directly by the native Sling Players or decoded and re-broadcasted by the Slinger server.

I have assigned an old laptop to run Slinger server and it works just fine since February 2023.

**Reply by GerryDazoo** [reply]
Read the networking guide to see what's possible. You don't have to run the server locally, you could just start up a copy on the same PC your using to run the player and access the slingbox remotely.

**Comment by waelelfeqy** [comment]
Thanks for your reply. That means that even wireshark can not trace the stream and capture the url; oh, well, at least I have a good use from my M1, not a brick. I bought a mini tv tuner from Amazon for 28 bucks, hooked it up to my composite on the back of my M1, and now I have a network TV tuner for 28 dollars !!. I don't know how to set up the remote though on my M1, can you help with that ? Otherwise I can't access your server remotely to change channels. Thanks

________________________________

**Reply by GerryDazoo** [reply]
I'm a little confused about your issue.
I think you're trying to access the slingbox directly from VLC. This will NOT work. If you want to use VLC you need to be running Slinger server that interworks an HTTP stream that VLC will eat with the proprietary SlingPlayer -> Slingbox protocol. There is no "URL" for direct Slingbox access.

**Comment by waelelfeqy** [comment]
Yes, exactly, I was trying to do that, just to avoid running a server on my main machine and accessing it from outside. But I'm still happy that I can use the M1 as a network tuner and continue to use it for the foreseeable future, even when ATSC 3.0 become available.
________________________________

**Comment by waelelfeqy** [comment]
No, I'm good now, but I can't figure out how to use the browser remote control to access the slingbox through the slinger server. How do I configure the remote control for M1 ?
________________________________
